discovery/gossiper.go

```diff
@@ -319,6 +320,117 @@ func (d *AuthenticatedGossiper) ProcessLocalAnnouncement(msg lnwire.Message,
 	return nMsg.err
 }
 
+// ChannelUpdateID is a unique identifier for ChannelUpdate messages, as
+// channel updates can be identified by the (ShortChannelID, Flags)
+// tuple.
+type ChannelUpdateID struct {
+	// ShortChannelID represents the set of data which is needed to
+	// retrieve all necessary data to validate the channel existence.
+	channelID lnwire.ShortChannelID
+
+	// Flags least-significant bit must be set to 0 if the creating node
+	// corresponds to the first node in the previously sent channel
+	// announcement and 1 otherwise.
+	Flags uint16
+}
+
+// deDupedAnnouncements de-duplicates announcements that have been
+// added to the batch. Internally, announcements are stored in three maps
+// (one each for channel announcements, channel updates, and node
+// announcements). These maps keep track of unique announcements and
+// ensure no announcements are duplicated.
+type deDupedAnnouncements struct {
+	// Channel announcements are identified by the short channel id field.
+	channelAnnouncements map[lnwire.ShortChannelID]lnwire.Message
+
+	// Channel updates are identified by the channel update id field.
+	channelUpdates map[ChannelUpdateID]lnwire.Message
+
+	// Node announcements are identified by node id field.
+	nodeAnnouncements map[*btcec.PublicKey]lnwire.Message
+}
+
+// Reset operates on deDupedAnnouncements to reset storage of announcements
+func (d *deDupedAnnouncements) Reset() {
+	// Storage of each type of announcement (channel anouncements, channel
+	// updates, node announcements) is set to an empty map where the
+	// approprate key points to the corresponding lnwire.Message.
+	channelAnnouncements := make(map[lnwire.ShortChannelID]lnwire.Message)
+	d.channelAnnouncements = channelAnnouncements
```

Comment:
Each of these mutations can be done in line, like so: 
```
d.channelAnnouncements = make(map[lnwire.ShortChannelID]lnwire.Message)
```

This'll shave a few lines of the total diff.